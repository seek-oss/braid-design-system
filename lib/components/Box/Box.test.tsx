import React from 'react';
import { Box } from '../';
import { render } from 'react-testing-library';
import forEachTheme from '../../test/utils/forEachTheme';

describe('Box', () => {
  forEachTheme(({ ThemeProvider }) => {
    test('No props', () => {
      const { container } = render(
        <ThemeProvider>
          <Box>Children</Box>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('All props', () => {
      const { container } = render(
        <ThemeProvider>
          <Box
            component="section"
            paddingTop="small"
            paddingBottom="small"
            paddingLeft="small"
            paddingRight="small"
            marginTop="small"
            marginBottom="small"
            marginLeft="small"
            marginRight="small"
            borderWidth="standard"
            borderRadius="standard"
            backgroundColor="card"
            className="GLOBAL_CLASS"
          >
            Children
          </Box>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });

    test('Unknown values', () => {
      const UnsafeBox: any = Box;

      const { container } = render(
        <ThemeProvider>
          <UnsafeBox
            paddingTop="UNKNOWN"
            paddingBottom="UNKNOWN"
            paddingLeft="UNKNOWN"
            paddingRight="UNKNOWN"
            marginTop="UNKNOWN"
            marginBottom="UNKNOWN"
            marginLeft="UNKNOWN"
            marginRight="UNKNOWN"
            borderWidth="UNKNOWN"
            borderRadius="UNKNOWN"
            backgroundColor="UNKNOWN"
          >
            Children
          </UnsafeBox>
        </ThemeProvider>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
