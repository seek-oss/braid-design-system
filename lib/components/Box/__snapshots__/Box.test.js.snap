// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Theme: jobStreet All props 1`] = `
<section
  class="GLOBAL_CLASS Box__root backgroundColor__card borderColor__default borderWidth__standard borderRadius__standard marginTop__small marginRight__small marginBottom__small marginLeft__small paddingTop__small paddingRight__small paddingBottom__small paddingLeft__small reset__section"
>
  Children
</section>
`;

exports[`Box Theme: jobStreet No props 1`] = `
<div
  class="Box__root borderColor__default"
>
  Children
</div>
`;

exports[`Box Theme: jobStreet Unknown values 1`] = `
<div
  class="Box__root borderColor__default"
>
  Children
</div>
`;

exports[`Box Theme: seekAnz All props 1`] = `
<section
  class="GLOBAL_CLASS Box__root backgroundColor__card borderColor__default borderWidth__standard borderRadius__standard marginTop__small marginRight__small marginBottom__small marginLeft__small paddingTop__small paddingRight__small paddingBottom__small paddingLeft__small reset__section"
>
  Children
</section>
`;

exports[`Box Theme: seekAnz No props 1`] = `
<div
  class="Box__root borderColor__default"
>
  Children
</div>
`;

exports[`Box Theme: seekAnz Unknown values 1`] = `
<div
  class="Box__root borderColor__default"
>
  Children
</div>
`;

exports[`Box Theme: seekAsia All props 1`] = `
<section
  class="GLOBAL_CLASS Box__root backgroundColor__card borderColor__default borderWidth__standard borderRadius__standard marginTop__small marginRight__small marginBottom__small marginLeft__small paddingTop__small paddingRight__small paddingBottom__small paddingLeft__small reset__section"
>
  Children
</section>
`;

exports[`Box Theme: seekAsia No props 1`] = `
<div
  class="Box__root borderColor__default"
>
  Children
</div>
`;

exports[`Box Theme: seekAsia Unknown values 1`] = `
<div
  class="Box__root borderColor__default"
>
  Children
</div>
`;
